File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldNotBePublic.vb
Namespace Tests.TestCases
    Public Class FieldShouldNotBePublic
        Public Shared Pi = 3.14
        Public Const Pi2 = 3.14
        Public ReadOnly Pi3 As Double = 3.14 ' Noncompliant {{Make 'Pi3' private.}}
'                       ^^^
        Public pi7, ' Noncompliant
            Pi4 As Double = 3.14, ' Noncompliant
            Pi6 As Double = 3.14 ' Noncompliant
        Private Pi5 As Double = 3.14
    End Class
End Namespace

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldNotBePublic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tests.TestCases
{
    public class FieldShouldNotBePublic
    {
        public static double Pi = 3.14;
        public const double Pi2 = 3.14;
        public readonly double Pi2 = 3.14; // Noncompliant {{Make 'Pi2' private.}}
//                             ^^^
        public double Pi3 = 3.14; // Noncompliant
        private double Pi4 = 3.14;
    }
}
